utils/src/treev2/quickPickWizard/GenericQuickPickStep.ts

```diff
@@ -72,7 +73,11 @@ export abstract class GenericQuickPickStep<TNode extends unknown, TContext exten
         let promptChoices: TNode[];
         if (directChoices.length === 0) {
             if (indirectChoices.length === 0) {
-                throw new NoResourceFoundError();
+                if (this.pickOptions.fallbackToAll) {
+                    promptChoices = children;

```

Comment:
I'm using it for QuickPickGroupStep, since we want to show all choices if none match the resource type we're looking for. This way it works regardless of the groupBy setting.

1. If grouped by type: a single group will match the `AzExtResourceType` we're looking for and it will be auto-picked
2. If not grouped by type: no group will match the `AzExtResourceType` we're looking for, thus fallback to show all children

Alternatively, we could check the groupBy setting and then only pass in `groupType` if the view is grouped by type. However, I wanted to avoid writing group-specific logic as much as I could.

https://github.com/microsoft/vscode-azuretools/blob/alex/tip-changes1/utils/src/treev2/quickPickWizard/experiences/appResourceExperience.ts#L24

Reply:
The scenario is inherently group-specific so I think we should change the behavior within `QuickPickGroupStep`.

Reply:
I think the code _should_ already handle that assuming that when grouped by type, the model item has `resourceType` set: https://github.com/microsoft/vscode-azuretools/blob/bb38f55322fd51ec15c432a2fe3fa0ab3ec9b732/utils/src/treev2/quickPickWizard/quickPickAzureResource/QuickPickGroupStep.ts#L34

Reply:
I think I attempted that, then out of laziness did it this way 😄. I will revisit doing it that way.

Reply:
Aha! Yeah, it'll require some changes in RG; it looks like that assumption about resource type is not currently valid, but I think it would be an easy thing to change.